### PR TITLE
Try making parenthesizeConstituentTypeOfUnionType a no-op

### PR DESCRIPTION
**Pull request by jakebailey**
Curious about https://github.com/microsoft/TypeScript/pull/57890#discussion_r1534857039 on DT

### CODE REVIEW

**Comment by jakebailey** [comment]
@typescript-bot run dt 
@typescript-bot perf test this

**Comment by jakebailey** [comment]
Actually not that bad

**Comment by weswigham** [comment]
Only 9 breaks on DT? Really not that bad.

File: tests/baselines/reference/extractFunction/extractFunction_VariableDeclaration_Writes_Let_TypeWithComments.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ function f() {
 
 function f() {
     let a = 1;
-    let x: (/*A*/ "a" /*B*/ | /*C*/ 'b' /*D*/) | undefined;
+    let x: /*A*/ /*A*/ "a" /*B*/ | /*C*/ 'b' /*D*/ | undefined;
```
[the remark this answers is not in the document]
Comment (jakebailey) [reply tests/baselines/reference/extractFunction/extractFunction_VariableDeclaration_Writes_Let_TypeWithComments.ts:26]: Any guesses as to where that logic might be?

[the remark this answers is not in the document]
Comment (weswigham) [reply tests/baselines/reference/extractFunction/extractFunction_VariableDeclaration_Writes_Let_TypeWithComments.ts:26]: Based on how `extractSymbol` works, probably some of the comment suppression/copying logic for `getSynthesizedDeepClone` or `getTypeDeepCloneUnionUndefined`.

**Pull request closed without merging**